src/abstractdataframe/abstractdataframe.jl

```diff
@@ -407,31 +408,36 @@ function StatsBase.describe(df::AbstractDataFrame; stats::Union{Symbol,AbstractV
     # An array of Dicts for summary statistics
     column_stats_dicts = map(columns(df)) do col
         if eltype(col) >: Missing
-            d = get_stats(collect(skipmissing(col)), stats)
+            t = collect(skipmissing(col))
+            d = get_stats(t, predefined_funs)
+            get_stats!(d, t, custom_funs)
         else
-            d = get_stats(col, stats)
+            d = get_stats(col, predefined_funs)
+            get_stats!(d, col, custom_funs)
         end
 
-        if :nmissing in stats
+        if :nmissing in predefined_funs 
             d[:nmissing] = eltype(col) >: Missing ? count(ismissing, col) : nothing
         end
 
-        if :first in stats
+        if :first in predefined_funs 
             d[:first] = isempty(col) ? nothing : first(col)
         end
-
-        if :last in stats
+        
+        if :last in predefined_funs
             d[:last] = isempty(col) ? nothing : last(col)
         end
 
         return d
     end
 
-    for stat in stats
+    for stat in ordered_names
         # for each statistic, loop through the columns array to find values
         # letting the comprehension choose the appropriate type
         data[stat] = [column_stats_dict[stat] for column_stats_dict in column_stats_dicts]
     end
+
+    # re-order columns according to the names from above 
```

Comment:
this comment is not needed here I think.

Reply:
fixed.